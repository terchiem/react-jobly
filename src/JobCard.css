.JobCard {
  position: relative;
  margin: 10px auto;
  padding: 10px 20px;
  border: 1px solid #ccc;
  box-shadow: 1px 1px 2px 0 rgba(0, 0, 0, 0.2);
  color: #111;
  background-color: #fff;
}

.JobCard-apply {
  position: absolute;
  bottom: 0;
  right: 0;
  margin: 20px;
  padding: 10px;
}

.JobCard-apply:hover {
  cursor: pointer;
}
