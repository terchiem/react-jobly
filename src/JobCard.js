import React from 'react';
import './JobCard.css';

/** Displays a job's details and a button to apply for the job */

function JobCard({ id, title, salary, equity, applied }) {

  /** Create an application for a user and a job */
  function applyForJob() {
    // TODO
  }

  return (
    <div className="JobCard">
      <h3>{title}</h3>
      <p>Salary: {salary}</p>
      <p>Equity: {equity}</p>
      <button className="JobCard-apply" onClick={applyForJob} disabled={applied}>
        {applied ? 'Applied' : 'Apply'}
      </button>
    </div>
  )
}

export default JobCard;