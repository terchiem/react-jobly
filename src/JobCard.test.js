import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import JobCard from './JobCard';

test('renders without crashing', () => {
  render(
    <MemoryRouter>
      <JobCard />
    </MemoryRouter>
  );
});

test('matches snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <JobCard />
    </MemoryRouter>
  );
  
  expect(asFragment()).toMatchSnapshot();
})