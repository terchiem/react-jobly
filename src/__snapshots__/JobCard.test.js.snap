// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card JobCard"
  >
    <h3 />
    <p>
      Salary: 
    </p>
    <p>
      Equity: 
    </p>
    <button
      class="JobCard-apply"
    >
      Apply
    </button>
  </div>
</DocumentFragment>
`;
